addons/mail/static/src/core/common/message.xml

```diff
@@ -121,7 +122,7 @@
     </t>
 
 <t t-name="mail.Message.actions">
-    <div t-if="props.hasActions and message.hasActions" class="o-mail-Message-actions"
+    <div t-if="props.hasActions and message.hasActions and ((!env.closeThread and !env.openThread) or !state.isEditing)" class="o-mail-Message-actions"
```

Comment:
knowledge can set `hasActions=False` to the message. You can introduce a new prop `messageActions=true/false` on component `Thread` to allow this customisation from `Thread` component which is what's used by knowledge